trin-beacon/src/storage.rs

```diff
@@ -515,7 +516,20 @@ impl BeaconStorage {
     /// Public method for looking up a content value by its content id
     pub fn lookup_content_value(&self, id: [u8; 32]) -> anyhow::Result<Option<Vec<u8>>> {
         let conn = self.sql_connection_pool.get()?;
-        lookup_content_value(id, conn)?
+        let mut query = conn.prepare(CONTENT_VALUE_LOOKUP_QUERY_BEACON)?;
+        let id = id.to_vec();
+        let result: Result<Vec<Vec<u8>>, ContentStoreError> = query
+            .query_map([id], |row| {
+                let row: Vec<u8> = row.get(0)?;
+                Ok(row)
+            })?
+            .map(|row| row.map_err(ContentStoreError::Rusqlite))
+            .collect();
+
+        match result?.first() {
```

Comment:
Wouldn't simple `result?.first()` work?

If not, I'm sure something like this should:
```rust
result?.first().map(|val| val.to_vec())
```

Reply:
Good find